extensions/replication/utils/SetupReplication.js

```diff
@@ -127,6 +127,7 @@ class SetupReplication extends BackbeatTask {
                 source.transport,
                 `${source.vault.host}:${source.vault.adminPort}`,
                 source.credentials, internalHttps),
+            // What is the proper way to use vault with the target?
```

Comment:
On S3C, it goes to the nginx proxy listening on port 9080, and gets redirected internally on the target to the appropriate Vault port.

On an open source deployment, I believe SetupReplication is not appropriate currently due to its reliance on Vault (unless `isExternal` which is meant for Zenko public cloud use cases), but we could consider modifying SetupReplication for open source use cases, which I think would be simply setting up the buckets and replication configurations without setting up roles.

Reply:
Hmm I am not sure that will work, the first parameter in the put bucket replication is the role https://docs.aws.amazon.com/cli/latest/reference/s3api/put-bucket-replication.html#examples

Reply:
I believe the role is not used in account credentials, or if it is, it's just to retrieve the account ID or something.

Also keep in mind that we hacked a little bit the role field to include both the source AND target replication roles (because we don't have a federated Vault, hence two different accounts on source and target). For this, we use two roles separated by a comma, instead of a single role.

In backbeat, the file extensions/replication/tasks/ReplicateObject.js contains some code related to parsing the role from the replication configuration.